core/src/main/java/hudson/model/UsageStatistics.java

```diff
@@ -193,18 +202,29 @@ public String getStatData() throws IOException {
 
     @Override
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
-        try {
-            // for backward compatibility reasons, this configuration is stored in Jenkins
-            Jenkins.getInstance().setNoUsageStatistics(json.has("usageStatisticsCollected") ? null : true);
-            return true;
+        try (BulkChange bc = new BulkChange(this)) {
+            req.bindJSON(this, json);
+            bc.commit();
+            Jenkins.get().setNoUsageStatistics(this.shouldCollectStatistics);
         } catch (IOException e) {
-            throw new FormException(e,"usageStatisticsCollected");
+            throw new FormException(e, "usageStatisticsCollected");
         }
+        return true;
+    }
+
+    public boolean getShouldCollectStatistics() {
+        return shouldCollectStatistics;
+    }
+
+    @DataBoundSetter
+    public void setShouldCollectStatistics(boolean shouldCollectStatistics) {
+        this.shouldCollectStatistics = shouldCollectStatistics;
```

Comment:
This does not write to `Jenkins#setNoUsageStatistics`, so I expect this to not work at all.

Please provide a test that this actually does what you claim.

Reply:
You are absolutely right! I would have thought the `configure` method gets called, but was wrong. I will move the logic to the `setter` and add tests. Regarding the whitespace changes, it was my IDE doing it on save. I disabled that, and will push without any whitespace changes. Thank you for the review!